src/HeaderPlugin.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Manages http headers.
+ */
+class HeaderPlugin implements Plugin
+{
+    /**
+     * @var array headers that have fixed value
+     */
+    private $headers = [];
+    private $removedHeader = [];
+
+    /**
+     * Sets a header with the given value to every requests.
+     *
+     * @param string $name     name of the header
+     * @param string $value    value of the header
+     * @param bool   $override By default if the header is already set on the request the value wont be replaced
+     *                         pass $override to true to make the header value the same for every request
+     */
+    public function withHeader($name, $value, $override = false)
```

Comment:
I called it withHeader and withoutHeader to make it look like the psr7 message interface. 

Do you prefer everything to be made at the instantiation?